node-plugin/lang/src/lib.rs

```diff
@@ -21,25 +19,31 @@ use fraction::{BigFraction, GenericFraction, Sign};
 use graphql_parser::query as q;
 use language::*;
 use num_bigint::BigUint;
-use std::{error, fmt};
+use std::{error, fmt, mem::transmute};
 
 pub use context::Context;
 // Hack for indexer selection
 pub use graphql_utils::QueryVariables;
 
-rental! {
-    mod rentals {
-        use super::*;
-        #[rental]
-        pub struct CostModelData {
-            text: String,
-            document: Document<'text>,
-        }
-    }
+pub struct CostModel {
+    // Rust does not have a memory model, nor does it have a proper `uintptr_t` equivalent. So a
+    // `*const u8` is used here, since the C99 standard explicitly allows casting from a C `char`
+    // the type of an object being accessed through a pointer (C99 §6.5/7). This assumes that all
+    // platforms being targeted will define C `char` as either a signed or unsigned byte.
+    document: *const u8,
+    // The `document` field uses references to the `text` field. In order to ensure safety, `text`
+    // must be owned by this struct and be dropped after `document`.
+    #[allow(dead_code)]
+    text: String,
 }
 
-pub struct CostModel {
-    data: rentals::CostModelData,
+unsafe impl Send for CostModel {}
+unsafe impl Sync for CostModel {}
+
+impl Drop for CostModel {
+    fn drop(&mut self) {
+        let _ = unsafe { Box::<Document>::from_raw(transmute(self.document())) };
```

Comment:
I don't know if it's unsafer to do this, but there's no need to use `transmute` here, this works as well:

```rust
Box::from_raw(self.document as *mut u8)
```

Reply:
Using `as` seems to be an overall better alternative to the `transmute` sledgehammer.